pkg/storage/chunk/util/parallel_chunk_fetch.go

```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/grafana/loki/pkg/storage/chunk"
 )
 
-const maxParallel = 1000
+const maxParallel = 150
```

Comment:
This should probably also be configurable if we're going to make such a drastic change

Reply:
I can't see a situation where above 150 is making sense. I'm actually stopping the bleed with this, we rarely download above 50 chunk in parallel but when we do we should not go higher than 150 per request otherwise we'll starve the process from goroutine. I'd prefer to avoid a configuration for this.

Reply:
To be clear: I'm not suggesting this value is incorrect or not optimal, but since we may want to tune this in the future it should at least be an (undocumented, perhaps) tuneable via config IMHO. Won't block the merge for this but if we're changing it now we may want to change it again.

Reply:
Sounds good I'll see how easy it is.

Reply:
Done